test/e2e/e2e_test.go

```diff
@@ -0,0 +1,106 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/serf/client"
+	"github.com/hashicorp/serf/cmd/serf/command"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+const (
+	webConsolePort = "4242"
+	serfPort       = "7373"
+	httpPort       = "8080"
+	rtmpPort       = "1935"
+)
+
+type catalystContainer struct {
+	testcontainers.Container
+	webConsole string
+	serf       string
+	http       string
+	rtmp       string
+}
+
+func startCatalyst(t *testing.T, ctx context.Context, hostname string, network string) *catalystContainer {
+	configAbsPath, err := filepath.Abs("../../config")
+	require.NoError(t, err)
+
+	var req = testcontainers.ContainerRequest{
+		Image:        "livepeerci/catalyst:pr-42",
+		ExposedPorts: []string{tcp(webConsolePort), tcp(serfPort), tcp(httpPort), tcp(rtmpPort)},
+		ShmSize:      1874000000,
```

Comment:
Should it be hardcoded value? This is decided by the shared memory buffer system has and could/should be computed at runtime

Reply:
yeah, I just took it from the Docker commands sent by @iameli . I think we can tweak it, decrease, or calculate at runtime.

Reply:
Removed `ShmSize`